src/main.rs

```diff
@@ -77,6 +78,21 @@ fn main() -> glib::ExitCode {
                 application_opts.test_dc = true;
             }
 
+            let maybe_client_id = dict.lookup::<String>("client-id").unwrap();
+            let maybe_client_secret = dict.lookup::<String>("client-secret").unwrap();
+
+            match (maybe_client_id, maybe_client_secret) {
+                (Some(client_id), Some(client_secret)) => {
+                    application_opts.client_id = client_id.parse::<i32>().unwrap();
+                    application_opts.client_secret = Cow::Owned(client_secret);
+                }
+                (None, None) => (),
+                _ => {
+                    log::error!("Both client-id and client-secret must be set together");
+                    return 1;
+                }
+            };
+
```

Comment:
Two last things: 

1. You don't need to parse client id as it is supposed to be an integer
2. You don't need `maybe` in the variables' names

```suggestion
            let client_id = dict.lookup::<i32>("client-id").unwrap();
            let client_secret = dict.lookup::<String>("client-secret").unwrap();

            match (client_id, client_secret) {
                (Some(client_id), Some(client_secret)) => {
                    application_opts.client_id = client_id;
                    application_opts.client_secret = Cow::Owned(client_secret);
                }
                (None, None) => (),
                _ => {
                    log::error!("Both client-id and client-secret must be set together");
                    return 1;
                }
            };

```

Reply:
I'm getting the following when removing the parsing:

```
86 |                     application_opts.client_id = client_id;
   |                     --------------------------   ^^^^^^^^^ expected `i32`, found `String`
```

I'm guessing the bindings aren't dynamic enough here to influence the type? Tried using `into()` as well but same thing.

Reply:
Please take a look at line 95

Reply:
ah, thanks, having a Sunday moment :D